index.d.ts

```diff
@@ -137,14 +42,844 @@ declare class Enquirer<T = object> extends EventEmitter {
+
+    swap(pos: number): void;
+
+    toChoice(element: string |
+      ((this: ArrayPrompt, arg: ArrayPrompt) => Promise<types.Choice>) |
+      Promise<types.Choice> |
+      types.Choice, i: number, parent?: types.Choice): Promise<types.Choice>;
+
+    toChoices(value: any, parent?: any): Promise<types.Choice[]>
+
+    toggle(choice: types.Choice, enabled: boolean): types.Choice | undefined;
+
+    up(): Promise<void>;
+  }
+
+  export namespace ArrayPrompt {
+    export type Question<T extends types.Answer, P extends ArrayPrompt<T> = ArrayPrompt<T>> = Prompt.Question.Base &
+    {
+      choices: (() => types.ChoiceInput[] | Promise<types.ChoiceInput[]>) | types.ChoiceInput[] | Promise<types.ChoiceInput[]>,
+      initial?: string | number | Array<string | number> | Record<string, any>,
+      autofocus?: number | string,
+      multiple?: boolean,
+      format?: (this: P, value: T) => any,
+      result?: (this: P, value: T) => any,
+    }
+  }
+  //#endregion
+
+  //#region Build-in prompts
+
+  export class AutoComplete<T extends types.Answer = string> extends Select {
+    constructor(question: AutoComplete.Question<T>)
+    complete(): Promise<void>
+    delete(): Promise<void>
+    deleteForward(): Promise<void>
+    // TODO: fix this
+    // pointer(): string
+    space(ch?: string | undefined): Promise<void>
+    suggest(input?: string, choices?: types.Choice[]): types.Choice[]
+  }
+  export namespace AutoComplete {
+    export type Question<
+      T extends types.Answer,
+      P extends AutoComplete = AutoComplete
+      > = ArrayPrompt.Question<T, P> & {
+        suggest?: (this: AutoComplete, input: string, choices: types.Choice[]) => types.Choice[] | Promise<types.Choice[]>;
+      }
+  }
+  export const autocomplete: AutoComplete
+
+  export class BasicAuth extends AuthPrompt { }
+  export const basicauth: BasicAuth
+
+  export class Confirm extends BooleanPrompt { }
+  export const confirm: Confirm
+
+  export class Editable extends Select { }
+  export const editable: Editable
+
+  export class Form extends Select { }
+  export const form: Form
+
+  export class Input extends StringPrompt {
+    altDown(): Promise<void>;
+
+    altUp(): Promise<void>;
+
+    completion(action: types.Action): Promise<void>;
+
+    prev(): void;
+
+    save(): void;
+
+    submit(): Promise<void>;
+  }
+  export const input: Input
+
+  export class Invisible extends StringPrompt {
+    format(): string;
+  }
+  export const invisible: Invisible
+
+  export class List extends Prompt<string[]> {
+    constructor(question: List.Question)
+    format(): string;
+    split(input?: string): string[];
+    submit(): Promise<void>;
+  }
+  export namespace List {
+    export type Question = Omit<Prompt.Question<string[]>, 'initial'> & {
+      initial?: string,
+      separator?: string | RegExp
+    }
+  }
+  export const list: List
+
+  export class MultiSelect extends Select {
+    constructor(question: MultiSelect.Question)
```

Comment:
`selected` is defined as `types.Choice | types.Choice[]`. For multiselect it should be:

```suggestion
    constructor(question: MultiSelect.Question)
    readonly selected: types.Choice[]
```

Reply:
We can definitely change that. I didn't do it because the code itself does not prevent this.